main.js

```diff
@@ -38,6 +40,14 @@ const updatePaddles = () => {
 };
 
 const updateBall = () => {
+  ball.x += ballSpeedHorizontal;
+  ball.y += ballSpeedVertical;
+  if (ball.x < ball.radius || ball.x > canvas.width / devicePixelRatio - ball.radius) {
+    ballSpeedHorizontal *= -1;
+  }
+  if (ball.y < ball.radius || ball.y > canvas.height / devicePixelRatio - ball.radius) {
+    ballSpeedVertical *= -1;
+  }
   ball.update();
```

Comment:
I like how you have all the update code in this function, although its different from how I usually do it.

I put in update into the classes assuming they would contain their own velocity.

If you keep the velocity here, then there is no need to have an update and this can get cleaned up. (along with the class methods).

```suggestion
```

Reply:
don't worry about this if it doesn't make sense to you. its OK to leave the line as is for now